src/main/scala/com/databricks/spark/redshift/RedshiftWriter.scala

```diff
@@ -85,28 +85,27 @@ private[redshift] class RedshiftWriter(
     val randomSuffix = Math.abs(Random.nextInt()).toString
     val tempTable = s"${table}_staging_$randomSuffix"
     val backupTable = s"${table}_backup_$randomSuffix"
+    log.info("Loading new Redshift data to: " + tempTable)
+    log.info("Existing data will be backed up in: " + backupTable)
 
     try {
-      log.info("Loading new Redshift data to: " + tempTable)
-      log.info("Existing data will be backed up in: " + backupTable)
-
       action(tempTable)
+
       if (jdbcWrapper.tableExists(conn, table)) {
-        conn.prepareStatement(s"ALTER TABLE $table RENAME TO $backupTable").execute()
+        conn.prepareStatement(
+          s"""
+             | BEGIN;
```

Comment:
Yeah, it passes the integration tests. This seems marginally safer than the approach that was there before because it eliminates the (remote) possibility of a fatal error occurring between the separate steps when renaming the tables.

I think that this approach is marginally cleaner, too, since it lets me change the scope of the try-finally block.